tests/tests/test_update_control.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2021 Northern.tech AS
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+
+import json
+import tempfile
+
+import pytest
+from testutils.common import User
+from testutils.infra.cli import CliTenantadm
+from testutils.infra.device import MenderDevice
+
+from .. import conftest
+from ..common_setup import enterprise_no_client
+from ..MenderAPI import Authentication, Deployments, DeviceAuthV2, image
+from .common_connect import wait_for_connect
+
+
+class TestUpdateControl:
+    def test_update_control(
+        self,
+        enterprise_no_client,
+        valid_image,
+    ):
+        """
+        Schedule an update with a pause in ArtifactInstall_Enter,
+        ArtifactReboot_Enter, and ArtifactCommit_Enter, then continue, after the
+        client has reported the paused substate back to the server, all the way
+        until the deployment is successfully finished.
+        """
+
+        u = User("", "bugs.bunny@acme.org", "whatsupdoc")
+        cli = CliTenantadm(containers_namespace=enterprise_no_client.name)
+        tid = cli.create_org("enterprise-tenant", u.name, u.pwd, "enterprise")
+        tenant = cli.get_tenant(tid)
+        tenant = json.loads(tenant)
+        ttoken = tenant["tenant_token"]
+
+        auth = Authentication(name="enterprise-tenant", username=u.name, password=u.pwd)
+        auth.create_org = False
+        auth.reset_auth_token()
+        devauth = DeviceAuthV2(auth)
+
+        enterprise_no_client.new_tenant_client("control-map-test-container", ttoken)
+        device = MenderDevice(enterprise_no_client.get_mender_clients()[0])
+        devauth.accept_devices(1)
+
+
+        deploy = Deployments(auth, devauth)
+
+        devices = list(
+            set([device["id"] for device in devauth.get_devices_status("accepted")])
+        )
+        assert 1 == len(devices)
+
+        wait_for_connect(auth, devices[0])
+
+        with tempfile.NamedTemporaryFile() as artifact_file:
+            created_artifact = image.make_rootfs_artifact(
+                "../"+valid_image,
+                conftest.machine_name,
+                "test-update-control",
+                artifact_file.name,
+            )
+
+            deploy.upload_image(artifact_file.name, description="control map update test")
+
+        deployment_id = deploy.trigger_deployment(
```

Comment:
This came up in the Slack channel: Can you, in addition to triggering the deployment, query the deployment afterwards, and make sure that the ID of the map it returns matches the ID of the deployment?

Reply:
Ref: https://northern-tech.slack.com/archives/C2D3HRU9M/p1623224323008800

Reply:
@kacf updated!